src/components/ArticleCard.jsx

```diff
@@ -1,23 +1,44 @@
 import Button from "@mui/material/Button";
-export default function ArticleCard({ article }) {
+import { GetSingleArticle } from "../apis/apis";
+import { useEffect, useState } from "react";
+import SingleArticle from "./SingleArticle";
+import { Link } from "react-router-dom";
+import { useParams } from "react-router-dom";
+import { Fade } from "react-awesome-reveal";
+export default function ArticleCard({ article, setViewSingleArticle, setId }) {
+  const handleClick = () => {
+    setViewSingleArticle(true);
+    setId(article.article_id);
+  };
+
   return (
-    <div className="article-card">
-      <div>
-        <p className="card-title"> Title </p>
-        <p className="card-sub-title">{article.title}</p>
+    <Fade>
+      <div className="article-card">
+        <div>
+          <p className="card-title"> Title </p>
+          <p className="card-sub-title">{article.title}</p>
+        </div>
+        <div>
+          <p className="card-title">Author: </p>
+          <p className="card-sub-title"> {article.author}</p>
+        </div>
+        <img src={article.article_img_url} alt="" />
+        <div>
+          <p className="card-title"> Topic</p>
+          <p className="card-sub-title">{article.topic}</p>
+          <Link to={`/articles/${article.article_id}`}>
```

Comment:
You have doubled up here by having both a Link to your articles/:article_id route and a ternary in Articles.jsx that is triggered by the onClick in your button. I would let your Link handle this functionality as the user will navigate to a new path and subsequently the SingleArticle component will render independently from the Articles component.